Makefile

```diff
@@ -81,24 +81,24 @@ MSBUILD = msbuild -verbosity:m -nologo
 # Enable 32 bit builds while generating the windows installer
 WIN32 = false
 
-# program targets
-VERSION     = $(shell git name-rev --name-only --tags --no-undefined HEAD 2>/dev/null || echo git-`git rev-parse --short HEAD`)
-
 # dependencies
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Darwin)
-os-dependencies = osx-dependencies
+TARGETPLATFORM = osx-x64
 else
-os-dependencies = linux-dependencies
+TARGETPLATFORM = linux-x64
```

Comment:
Do we want to handle bsd here? Either detect it or let the user supply a parameter for TARGETPLATFORM?

Reply:
I've been assuming that BSD can be covered under `linux-generic`, but you're right that we should at least make sure that it won't autodetect to `linux-x64`. @jbeich: Can you offer any insight here?

Reply:
also `unix-generic` maybe is a better name?

Reply:
Sounds reasonable, this would apply to macOS too then.

Reply:
Something like this, but not sure we need `ifndef (TARGETPLATFORM)`
http://www.gnu.org/software/make/manual/make.html#Overriding

```
# dependencies
ifndef (TARGETPLATFORM)
UNAME_S := $(shell uname -s)
UNAME_M := $(shell uname -m)
ifeq ($(UNAME_S),Darwin)
TARGETPLATFORM = osx-x64
else
ifeq ($(UNAME_M),x86_64)
TARGETPLATFORM = linux-x64
else
TARGETPLATFORM = linux-generic
endif
endif
endif
```

Reply:
`unix-generic` should work but don't remove `ifndef (TARGETPLATFORM)`. DragonFly and NetBSD may return `x86_64` while Gentoo Linux may want to avoid pre-built binaries.